zipper/protocols/irondb/irondb_group.go

```diff
@@ -0,0 +1,682 @@
+		limiter:              limiter,
+		logger:               logger,
+	}
+
+	return c, nil
+}
+
+func New(logger *zap.Logger, config types.BackendV2, tldCacheDisabled bool) (types.BackendServer, merry.Error) {
+	if config.ConcurrencyLimit == nil {
+		return nil, types.ErrConcurrencyLimitNotSet
+	}
+	if len(config.Servers) == 0 {
+		return nil, types.ErrNoServersSpecified
+	}
+	l := limiter.NewServerLimiter([]string{config.GroupName}, *config.ConcurrencyLimit)
+
+	return NewWithLimiter(logger, config, tldCacheDisabled, l)
+}
+
+func (c *IronDBGroup) Children() []types.BackendServer {
+	return []types.BackendServer{c}
+}
+
+func (c IronDBGroup) MaxMetricsPerRequest() int {
+	return c.maxMetricsPerRequest
+}
+
+func (c IronDBGroup) Name() string {
+	return c.groupName
+}
+
+func (c IronDBGroup) Backends() []string {
+	return c.servers
+}
+
+func processFindErrors(err error, e merry.Error, stats *types.Stats, query string) merry.Error {
+	stats.FindErrors++
+	if merry.Is(err, types.ErrTimeoutExceeded) {
+		stats.Timeouts++
+		stats.FindTimeouts++
+	}
+	if e == nil {
+		e = merry.Wrap(err).WithValue("query", query)
+	} else {
+		e = e.WithCause(err)
+	}
+	return e
+}
+
+func processRenderErrors(err error, e merry.Error, stats *types.Stats, query string) merry.Error {
+	stats.RenderErrors++
+	if merry.Is(err, types.ErrTimeoutExceeded) {
+		stats.Timeouts++
+		stats.RenderTimeouts++
+	}
+	if e == nil {
+		e = merry.Wrap(err).WithValue("query", query)
+	} else {
+		e = e.WithCause(err)
+	}
+	return e
+}
+
+func (c *IronDBGroup) Fetch(ctx context.Context, request *protov3.MultiFetchRequest) (*protov3.MultiFetchResponse, *types.Stats, merry.Error) {
+	logger := c.logger.With(zap.String("type", "fetch"), zap.String("request", request.String()))
+	stats := &types.Stats{}
+
+	pathExprToTargets := make(map[string][]string)
+	for _, m := range request.Metrics {
+		targets := pathExprToTargets[m.PathExpression]
+		pathExprToTargets[m.PathExpression] = append(targets, m.Name)
+	}
+
+	var r protov3.MultiFetchResponse
+	var e merry.Error
+
+	start := request.Metrics[0].StartTime
+	stop := request.Metrics[0].StopTime
+	step := c.graphiteRollup
+	count := int64((stop-start)/step) + 1
+	maxCount := request.Metrics[0].MaxDataPoints
+	if count > maxCount && maxCount > 0 {
+		count = maxCount
+		step = adjustStep(start, stop, maxCount, step)
+	}
+	if count <= 0 {
+		logger.Fatal("stop time should be less then start",
+			zap.Int64("start", start),
+			zap.Int64("stop", stop),
+		)
+	}
+	for pathExpr, targets := range pathExprToTargets {
+		for _, target := range targets {
+			logger.Debug("got some target to query",
+				zap.Any("pathExpr", pathExpr),
+				zap.String("target", target),
+				zap.Int64("start", start),
+				zap.Int64("stop", stop),
+				zap.Int64("count", count),
+				zap.Int64("period", step),
+			)
+
+			var query string
+			if strings.HasPrefix(target, "seriesByTag") {
+				query = target[len("seriesByTag(") : len(target)-1]
```

Comment:
Maybe, better use a const for `len("seriesByTag(")`. Otherwise, it will allocate string and int for each target